stl/inc/xstring

```diff
@@ -3165,21 +3149,24 @@ public:
                 auto&& _Right_alproxy = _GET_PROXY_ALLOCATOR(_Alty, _Right_al);
                 _Container_proxy_ptr<_Alty> _New_proxy(_Right_alproxy, _Leave_proxy_unbound{}); // throws
 
-                if (_Right._Mypair._Myval2._Large_string_engaged()) {
-                    const auto _New_size     = _Right._Mypair._Myval2._Mysize;
-                    const auto _New_capacity = _Calculate_growth(_New_size, 0, _Right.max_size());

```

Comment:
This part was broken and can cause allocation smaller than `BUF_SIZE`. (e.g. consider when `large_string_engaged()`, but `size() == 0`)